bin/run-test.sh

```diff
@@ -1,18 +1,56 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
+# Usage: -c to report coverage
+
+while true; do
+  case $1 in
+    -c)
+      cover=1
+      ;;
+
+    *)
+      break
+  esac
+
+  shift
+done
+
+# Lint
 jshint . || exit 1
+
+# Install framework deps
 for dir in test/hooks/fixtures/*/ ;
 do
   echo -en "travis_fold:start:npm_install_${dir}\\r" | tr / _
   echo "npm install in ${dir}"
-  cd ${dir} && npm install && cd -
+  cd "${dir}" && npm install && cd - || exit
   echo -en "travis_fold:end:npm_install_${dir}\\r" | tr / _
 done
-istanbul test $(npm bin)/_mocha -- test test/hooks --timeout 4000 --R spec || exit 1
+
+# Get test/coverage command
+function command {
+  C="istanbul test"
+  if [ "$cover" ]; then
+    C="istanbul cover --dir ./coverage/$RANDOM"
```

Comment:
It would be better to avoid random number collisions. Use this instead:

Initialize `counter` globally to 0.
```bash
C="istanbul cover --dir ./coverage/$((counter++))"
```